### ISSUE TITLE
time series kfold

### ISSUE THREAD

**Issue by nasospat** [pycaret/pycaret#79]
Is it possible to use k-fold for time series as skleanr uses ?
 https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html

**Comment by ngupta23** [comment]
Just my 2 cents...The models in PyCaret don't seem suitable for use with correlated data. What model do you wan to use with your time series data?

**Comment by nasospat** [comment]
@ngupta23 this link may help you underastand why in some cases we need TimeSeriesSplit
https://www.kaggle.com/kashnitsky/correct-time-aware-cross-validation-scheme

**Comment by ngupta23** [comment]
I understand the concept of time series cross validation, but that is only useful if the algorithms can take advantage of the serial correlation in the data. As far as I can see, none of the algorithms in the PyCaret framework do that. 

Which particular algorithm in PyCaret did you want to use for time series analysis?  If you are just using Logistic Regression, then you don't need the time series splitter since logistic regression is only taking one time point for making the prediction without consideration for any of the previous time points.

**Comment by nasospat** [comment]
Training and evaluating machine learning models require a training set and a test set. In most cases, train and test splitting is done randomly by taking a percent of the data as test data, unseen by the model and using the rest for training.  There are some cases when dealing with time-related and dynamically changing environments, where the characteristics of the environment change throughout time, it is best to use time-based splitting to provide statistically robust model evaluation and best simulate real-life scenarios. For that reason sklearn has made a time-based cross validation.  Most of the supervised learning algorithms that Pycaret supports are suitable for that technique.

**Comment by ngupta23** [comment]
I think there are 2 things here - time series CV splitting and the model actually taking advantage of the serial data. The former will only ensure that the future correlated data points don't leak into the test set. The latter will require models like ARIMA, VARS, RNNs (and its variants), etc which are not supported by pycaret at this time.

In any case, I will work on a pull request to pass a custom Cross Validation object which would be a useful feature in itself. You could be used for Time Series Split if you wish as well.

**Comment by pycaret** [comment]
Hi All,

We will be introducing this functionality in the next release `2.2` (exp by end of Sep).

We are introducing a few new params in the `setup` function that allows you to pick the fold strategy and shuffle. By allowing `shuffle` to be disabled during `train_test_split` and the ability to choose `timeseries` as a fold_strategy instead of `kfold` or `stratifiedkfold`, this should satisfy the need.

@Yard1 FYI. You have already taken this change.

**Comment by pycaret** [comment]
@nasospat @ngupta23 

This issue has been fixed in the `pycaret-nightly` release. Please install pycaret-nightly using `pip install pycaret-nightly`. Feel free to open a new issue, in case the problem persists. This fix will be published to PyPi as `pycaret==2.2.0` on October 28, 2020.
